src/dbus/clientprivate.cpp

```diff
@@ -134,11 +146,15 @@ void Ngf::ClientPrivate::ngfdStatus(const QString &service, const QString &arg1,
             QDBusInterface *iface = m_iface;
             m_iface = 0;
             // All currently active events are invalid, so clear event list
-            m_events.clear();
-            // Signal that we are disconnected
-            emit q_ptr->connectionStatus(false);
-            if (iface)
+            while (!m_events.isEmpty()) {
```

Comment:
And if it's a problem that `m_events` might be accessed by multiple threads, you could:

```
QList<Event*> tmp;
std::swap(m_events, tmp);
for (auto &event: tmp) {
    delete event;
}
```

This way, `m_events` will only contain either all alive items or be empty.